UX Multichain: Updated header to use picker network in popup view

NidhiKJha opened the pull request:
This PR is to use a network picker without a label for the header in the Popup view rather than avatar network

Fixes: [MetaMask-planning/issues/700](https://github.com/MetaMask/MetaMask-planning/issues/700)

## Screenshots/Screencaps

### Before
![Screenshot 2023-06-08 at 4 24 08 PM](https://github.com/MetaMask/metamask-extension/assets/39872794/6d6fc189-ec4b-47bf-800e-54f67ea0d872)


### After
![Screenshot 2023-06-08 at 4 23 28 PM](https://github.com/MetaMask/metamask-extension/assets/39872794/76d6ceab-05bb-4bb1-bfd2-8235da244fea)


## Manual Testing Steps
- In the popup view in the unlocked state the network list is rendered with a picker network without label.


## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

NidhiKJha committed:
updated avatar network to use picker network in popup view

NidhiKJha committed:
removed unused locale

NidhiKJha committed:
Merge branch 'develop' into fix-700

darkwing reviewed (approved):
Looks good but is there too much space between the avatar and the "v"?
<img width="218" alt="SCR-20230608-hurt" src="https://github.com/MetaMask/metamask-extension/assets/46655/a798d12c-75ab-4f89-b9b1-9444fa2284f7">

darkwing reviewed (dismissed):
Looks good but is there too much space between the avatar and the "v"?
<img width="218" alt="SCR-20230608-hurt" src="https://github.com/MetaMask/metamask-extension/assets/46655/a798d12c-75ab-4f89-b9b1-9444fa2284f7">

NidhiKJha committed:
fixed the extra gap

NidhiKJha commented:
> Looks good but is there too much space between the avatar and the "v"? 

Nice catch! Fixed it here https://github.com/MetaMask/metamask-extension/pull/19514/commits/a79f41433c4b88cb8a056f13f0f235a976426ca7. In my previous approach, the label tag was not a hidden element and hence it was taking its space in the picker network. So, I reverted the change in Picker Network to keep the label as a required prop and controlled the style via css here

NidhiKJha committed:
fixed css

NidhiKJha committed:
fixed lint error

vthomas13 reviewed (dismissed):
Looks great, I really like this change

georgewrmarshall commented on ui/components/multichain/app-header/app-header.scss:
```diff
@@ -33,8 +33,8 @@
     }
 
-    &--avatar-network {
-      padding: 0; // TODO: Remove once https://github.com/MetaMask/metamask-extension/pull/17006 is merged
```
Apologies, I closed the PR but did not merge it you may still need this. Once the component-library components are using the new Box component we can remove this

georgewrmarshall replied:
Actually now it's merged you could update the Box component in the PickerNetwork to use the component-library version and you could remove this.

NidhiKJha replied:
I am not sure why do we need to update the Picker network here. I don't need the padding so I removed it. CSS update is just to hide the label tag. 🤔

georgewrmarshall commented on ui/components/multichain/app-header/app-header.scss:
```diff
@@ -33,8 +33,8 @@
-    &--avatar-network {
-      padding: 0; // TODO: Remove once https://github.com/MetaMask/metamask-extension/pull/17006 is merged
+    &--avatar-network p {
+      display: none;
```
So I think this will have some accessibility concerns. But I see that `label` is a required prop for PickerNetwork I think we may need to adjust the `PickerNetwork` component slightly.

NidhiKJha replied:
Is it fine that we resolved it with aria-label now or does this need any update in Picker Network as well?

georgewrmarshall replied:
I'm actually not sure but I think the aria-label should be fine 👍

georgewrmarshall commented on ui/components/multichain/app-header/app-header.js:
```diff
@@ -195,15 +193,12 @@ export const AppHeader = ({ location }) => {
                       className="multichain-app-header__contents--avatar-network"
                       ref={menuRef}
                       as="button"
-                      aria-label={t('networkMenu')}
```
I think we still need this aria-label as there is no text describing what the button does for screen readers?

NidhiKJha replied:
Yup! Updated https://github.com/MetaMask/metamask-extension/pull/19514/commits/84530cab2ebdf1edebbfae3bc1719b32efc42e63

georgewrmarshall commented on ui/components/multichain/app-header/app-header.js:
```diff
@@ -23,10 +23,8 @@ import {
   BLOCK_SIZES,
   DISPLAY,
```
While we are here we could update these deprecated consts to enums

NidhiKJha replied:
Updated https://github.com/MetaMask/metamask-extension/pull/19514/commits/84530cab2ebdf1edebbfae3bc1719b32efc42e63

georgewrmarshall reviewed (commented):
Hey @NidhiKJha, this looks fantastic but I think there are some accessibility concerns that may need addressing

NidhiKJha committed:
updated aria label

NidhiKJha committed:
Merge branch 'develop' into fix-700

darkwing reviewed (approved):
Awesome!
